flytekit/remote/remote.py

```diff
@@ -1717,6 +1726,135 @@ def generate_console_http_domain(self) -> str:
+        )
+        if dry_run:
+            output("\n Dry Run enabled. Workflow will not be registered and or executed.", fg="yellow")
+            return wf
+
+        unique_fingerprint = f"{from_date}-{to_date}-{launchplan}-{launchplan_version}"
+        h = hashlib.md5()
+        h.update(unique_fingerprint.encode("utf-8"))
+        unique_fingerprint_encoded = base64.urlsafe_b64encode(h.digest()).decode("ascii")
+        if not version:
+            version = unique_fingerprint_encoded
+        ss = SerializationSettings(
+            image_config=ImageConfig.auto(),
+            project=project,
+            domain=domain,
+            version=version,
+        )
+        remote_wf = self.register_workflow(wf, serialization_settings=ss)
+
+        if no_execute:
+            return remote_wf
+
+        return self.execute(remote_wf, inputs={}, project=project, domain=domain, execution_name=execution_name)
+
+    @staticmethod
+    def create_backfiller(
+        start_date: datetime,
+        end_date: datetime,
+        for_lp: typing.Union[LaunchPlan, FlyteLaunchPlan],
+        parallel: bool = False,
+        per_node_timeout: timedelta = None,
+        per_node_retries: int = 0,
+        output: typing.Any = ignore_extra_print,
+    ) -> WorkflowBase:
+        """
+        Generates a new imperative workflow for the launchplan that can be used to backfill the given launchplan.
+        This can only be used to generate  backfilling workflow only for schedulable launchplans
+
+        the Backfill plan is generated as (start_date - exclusive, end_date inclusive)
+        """
+        if not for_lp:
+            raise ValueError("Launch plan is required!")
+
+        if start_date >= end_date:
+            raise ValueError(
+                f"for a backfill start date should be earlier than end date. Received {start_date} -> {end_date}"
+            )
+
+        schedule = for_lp.entity_metadata.schedule if isinstance(for_lp, FlyteLaunchPlan) else for_lp.schedule
+
+        if schedule is None:
+            raise ValueError("Backfill can only be created for scheduled launch plans")
+
+        if schedule.cron_schedule is not None:
+            cron_schedule = schedule.cron_schedule
+        else:
+            raise NotImplementedError("Currently backfilling only supports cron schedules.")
+
+        output(f"Generating backfill from {start_date} -> {end_date}. Parallel?[{parallel}]", fg="yellow")
+        wf = ImperativeWorkflow(name=f"backfill-{for_lp.name}")
+        date_iter = croniter(cron_schedule.schedule, start_time=start_date, ret_type=datetime)
+        prev_node = None
+        while True:
+            next_start_date = date_iter.get_next()
+            if next_start_date >= end_date:
+                break
+            next_node = wf.add_launch_plan(for_lp, t=next_start_date)
+            next_node = next_node.with_overrides(
+                name=f"b-{next_start_date}", retries=per_node_retries, timeout=per_node_timeout
```

Comment:
what's the `b-` prefix for?

Reply:
just to ensure the node begins with a string (backfill)

Reply:
can we make this more explicit, like `"backfill-{next_start_date}"`?